src/pages/todo/todo.tsx

```diff
@@ -3,41 +3,45 @@ import Row from "react-bootstrap/Row";
 import Col from "react-bootstrap/Col";
 import Button from "react-bootstrap/Button";
 import Alert from "react-bootstrap/Alert";
+import Spinner from 'react-bootstrap/Spinner';
 import "react-datepicker/dist/react-datepicker.css";
 
-import { TodoForm } from "../../components/todoForm/todoForm";
-import { info_obj } from "../../interface/todoInterface";
-import { TodoTable } from "../../components/todoTable/todoTable";
+import { TodoForm } from "../../components/todoForm/";
+import { DefaultInfo} from "../../interface/";
+import { TodoTable } from "../../components/todoTable/";
+import { addTask, getAllTask, deleteTask, redirectToHome } from '../../utils/';
+import { useTokenContext } from '../../context/';
 
 declare function require(name:string);
 var I18n = require('react-redux-i18n').I18n;
 
 const Todo: React.FC = () =>{
-
-    const [info, setInfo] = useState<info_obj>({description: "", category: "css", content: "", deadline: new Date(), id: Math.floor(Math.random() * 1000)});
+    const {token} = useTokenContext();
+    const [info, setInfo] = useState(DefaultInfo);
     const [selectData, setSelectData] = useState([]) as any;
     const [data, setData] = useState([]) as any;
+
     const [noteShow, setNoteShow] = useState(false);
     const [validShow, setValidShow] = useState(false);
     const [note, setNote] = useState("add/delete success");
     const [todoShow, setToDoShow] = useState(false);
+    const [addNewLoading, setAddNewLoading] = useState(false);
+    const [deleteLoading, setDeleteLoading] = useState(false);
 
     useEffect(() => {

```

Comment:
let's create custom hooks to handle these API calls instead of calling `useEffect` in each component.

```suggestion
export const useAllTasks = (): Task[] => {
   ...
}

```